kyuubi-server/src/main/scala/yaooqinn/kyuubi/operation/KyuubiClientOperation.scala

```diff
@@ -452,43 +321,21 @@ class KyuubiOperation(session: KyuubiSession, statement: String) extends Logging
     }
   }
 
-  private def onStatementError(id: String, message: String, trace: String): Unit = {
-    error(
-      s"""
-         |Error executing query as ${session.getUserName},
-         |$statement
-         |Current operation state ${this.state},
-         |$trace
-       """.stripMargin)
-    setState(ERROR)
+  override protected def onStatementError(id: String, message: String, trace: String): Unit = {
+    super.onStatementError(id, message, trace)
     KyuubiServerMonitor.getListener(session.getUserName)
       .foreach(_.onStatementError(id, message, trace))
   }
 
-  private def cleanup(state: OperationState) {
-    if (this.state != CLOSED) {
-      setState(state)
-    }
-    val backgroundHandle = getBackgroundHandle
-    if (backgroundHandle != null) {
-      backgroundHandle.cancel(true)
-    }
+  override protected def cleanup(state: OperationState) {
+    super.cleanup(state)
     if (statementId != null) {
       sparkSession.sparkContext.cancelJobGroup(statementId)
     }
   }
-
-  def isTimedOut: Boolean = {
-    if (operationTimeout <= 0) {
-      false
-    } else {
-      // check only when it's in terminal state
-      state.isTerminal && lastAccessTime + operationTimeout <= System.currentTimeMillis()
-    }
-  }
 }
 
-object KyuubiOperation {
+object KyuubiClientOperation {
```

Comment:
move this to object IKyuubiOperation or package object

Reply:
done.